client_plugin/drivers/shared/dockerops/dockerops.go

```diff
@@ -142,13 +175,165 @@ func (d *DockerOps) VolumeRemove(volName string) error {
 }
 
 // StartSMBServer - Start SMB server
-func (d *DockerOps) StartSMBServer(volName string) bool {
-	log.Errorf("startSMBServer to be implemented")
-	return true
+func (d *DockerOps) StartSMBServer(volName string) (int, string, bool) {
+	var service swarm.ServiceSpec
+	var options dockerTypes.ServiceCreateOptions
+
+	// Name of the service
+	service.Name = volName + serviceNamePostfix
```

Comment:
Add a random number to service name. This will be great precaution against orphan Samba services running.

If an orphan is running for the same volume name, and someone mounts from global refcount 0 -> 1, new samba service creation will fail because of service name conflict.

Adding a random number will avoid a name conflict.